server/src/LspServer.ts

```diff
@@ -149,25 +136,78 @@ export class LspServer {
       // Register for all configuration changes.
       await this.connection.client.register(DidChangeConfigurationNotification.type, undefined);
     }
+
+    const [, contextInfo] = await Promise.all([this.prepareRpcServer(), this.prepareDestination()]);
+
+    const initTime = performance.now() - this.initStart;
+    this.logStartupInfo(contextInfo, initTime, this.initDestinationAttempt, this.initDbtRpcAttempt);
+  }
+
+  async prepareDestination(): Promise<ContextInfo> {
+    this.initDestinationAttempt++;
+
+    const bigQueryContextInfo = await BigQueryContext.createContext(this.yamlParser);
+    if (this.contextFirstInitPromiseResolve) {
+      this.contextFirstInitPromiseResolve(true);
+    }
+
+    if (bigQueryContextInfo.isOk()) {
+      this.bigQueryContext = bigQueryContextInfo.value;
+      return this.bigQueryContext.contextInfo;
+    }
+
+    return this.showErrorDialog<ContextInfo>(
```

Comment:
`showErrorDialog` method became very complicated. It says about error, but it can show warning. It has retry logic but why do we need retry for warning? Initially I did retry logic only for problems that can be resolved by user when they see error message. Can we simplify it (probably split to 2 methods) and do retries only when it can help user? Or maybe get rid of retries at all because now I'm not sure that they work correctly.